scripts/ci/setup-dev-sso.sh

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env bash
+
+user_help () {
+    echo "Deploy in cluster keycloak and configure registration service to use it."
+    echo "options:"
+    echo "-sn, --sso-ns  Builds and pushes the operator to quay"
+    echo "-h,  --help              To show this help text"
+    echo ""
+    exit 0
+}
+
+read_arguments() {
+    if [[ $# -lt 2 ]]
+    then
+        echo "There are missing parameters"
+        user_help
+    fi
+
+    while test $# -gt 0; do
+           case "$1" in
+                -h|--help)
+                    user_help
+                    ;;
+                -sn|--sso-ns)
+                    shift
+                    DEV_SSO_NS=$1
+                    shift
+                    ;;
+                *)
+                   echo "$1 is not a recognized flag!" >> /dev/stderr
+                   user_help
+                   exit -1
+                   ;;
+          esac
+    done
+}
+check_commands()
+{
+    for cmd in "$@"
+    do
+        check_command "$cmd"
+    done
+}
+
+check_command()
+{
+    command -v "$1" > /dev/null && return 0
+
+    printf "please install '%s' before running this script\n" "$1"
+    exit 1
+}
+
+read_arguments $@
+
+if [[ -n "${CI}" ]]; then
+    set -ex
+else
+    set -e
+fi
+
+
+check_commands yq oc base64 openssl
+
+# Install rhsso operator
+SUBSCRIPTION_NAME=${DEV_SSO_NS}
+printf "installing RH SSO operator\n"
+    INSTALL_RHSSO="apiVersion: v1
+kind: Namespace
```

Comment:
I'd suggest moving those manifests in a local `manifests` folder and use `kustomize` for building and applying them. To me, this would make the script way more readable and maintainable.

For injecting the variables we may use the `configMapGenerator` and `replacements` features by kustomize.

Reply:
@filariow thanks for your suggestion. 

IMO there are two parts here:
1. decoupling the yaml CRs from the bash script and use `envsubst` to pass the secret and namespace from the scripts/make targets , which will make it cleaner , even thought I don't think we will change those yaml files/scripts so often once they are stable.

2.  using kustomize to apply the manifests, TBH I don't think this is strictly necessary , we don't have different overalys to apply , the configuration is always the same atm, and `oc apply` would do just fine right ? But maybe I'm missing the benefit of using kustomize in this situation.

In brief, I agree on the decoupling of the yaml files from the bash script, but not sure we need `kustomize` for now, using `oc` to apply the manifest should work just fine for now. WDYT ?

Reply:
> For injecting the variables we may use the configMapGenerator and replacements features by kustomize.

TBH I find kustomize a bit limited when comes to injecting variables from scripts, and I think it's consider an anti-pattern, but I'm not an expert here.

Reply:
I agree with your analysis, and yes, maybe `kustomize` is not strictly required.

I'd propose `kustomize` over the `envsubst` approach as I find it a bit cleaner and, as `kustomize` is the default technology used for baking and deploying manifests in the whole project, using it also here -seems to me- will provide an overall smoother experience to users.

If I'm not mistaken, we can use the script for generating a `.gitignored` overlay. The overlay will just contain a configuration file and the kustomization file with `base`, `configMapGenerator`, and `replacements`. This approach should not be considered an anti-pattern, but I'm not 100% sure.

At the end, it's basically the same of using `envsubst`. The main difference is that with the `kustomize` based approach, we will persist the configuration on disk. `envsubst` will be completely in memory, easier to implement, and -maybe- a bit harder to debug and inspect.

Btw, it seems to me it's more a team preference than a technical requirement 😄

Reply:
yeah, I've proposed to keep it simple using envsubst because I'm less familiar on how to do it with kustomize / configMapGenerator and replacements 🤷‍♂️ . Do you have an example ?

Reply:
meantime I've decoupled the yaml files from the bash script in https://github.com/codeready-toolchain/toolchain-cicd/pull/113/commits/07761694b9f83d5e089c2a366f10a487e7934216 and used `envsubst`.

Reply:
From the top of my mind, this may be a starting point: https://github.com/konflux-workspaces/workspaces/blob/main/server/config/server/kustomization.yaml

Reply:
https://github.com/konflux-workspaces/workspaces/blob/main/server/config/default/kustomization.yaml